### PR TITLE
Add auto-knockout task

### LINKED ISSUE #30: Automatic knock-out logging
Need to log knock-outs so the podium message can correctly exclude people who got knocked out.

Should include a knock-out message and a "saved by mulligan" message.

### PR DESCRIPTION
**Pull request by benz145**
## Summary
- Add scheduled auto-knockout handling with Discord notifications and medal refreshes after mulligan/knockout events.
- Evaluate green-week state for an explicit challenge week so historical auto-knockout logic can run deterministically.
- Exclude knocked-out challengers from medal eligibility and handle missing week medal winners without crashing.

## Test plan
- Not run in this session.
- Added focused tests for auto-knockout behavior and medal reconciliation edge cases.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/auto_knockout.py [hunk 1]
```diff
@@ -0,0 +1,849 @@
+from dataclasses import dataclass
+from datetime import date, datetime, time, timedelta
+import logging
+
+
+AUTO_KNOCKOUT_RUNS_TABLE = """
```
Comment (trescenzi) [inline src/auto_knockout.py]: This should be stateless. We shouldn't need to track when this has run

File: src/auto_knockout.py [hunk 2]
```diff
@@ -0,0 +1,849 @@
...
+        action = "mulligan"
+    else:
+        action = "knockout"
+
+    return AutoKnockoutDecision(
+        action=action,
+        required_checkins=required_checkins,
+        checkin_count=checkin_count,
+        remaining_possible_days=remaining_days,
+    )
```
Comment (trescenzi) [inline src/auto_knockout.py]: This is probably all the code that's needed minus getting the values above

File: src/green.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import random
 
 
-def number_of_non_green_weeks_before_this_one(challenge_id):
+def number_of_non_green_weeks_before_week(challenge_id, week_start):
```
Comment (trescenzi) [inline src/green.py]: Why are we making changes here?

**Review by trescenzi — CHANGES_REQUESTED** [review]
Honestly there's too much here for me to really go through. All we need is:

- Pull the checkins per challenger for the last week
- If they have less than 2 check if they've mulliganed
- If they haven't mulliganed mulligan them inserting a checkin for the first day they didn't checkin last week and storing that information in challenger_challenges.mulligan
- If they have mulliganed knock them out by setting challenger_challenges.knocked_out = true

There's a ton here that's not needed I'd just give the above to your agent and see what it does with it

**Review by trescenzi — APPROVED** [review]
generally seems ok. excited for this to finally be a thing!
